compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -536,15 +534,15 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             moved_place,
             DescribePlaceOpt { including_downcast: true, including_tuple_field: true },
         ) {
-            Some(name) => (format!("`{name}`"), format!("`{name}` ")),
-            None => ("the variable".to_string(), String::new()),
+            Some(name) => (&*format!("`{name}`"), format!("`{name}` ")),
+            None => ("the variable", String::new()),
         };
         let path = match self.describe_place_with_options(
             used_place,
             DescribePlaceOpt { including_downcast: true, including_tuple_field: true },
         ) {
-            Some(name) => format!("`{name}`"),
-            None => "value".to_string(),
+            Some(name) => &format!("`{name}`"),
+            None => "value",
```

Comment:
Could you break `let name_bt = format!("`{name}`");` out and then just use `&name_bt` in all these places? Since the same formatting happens three times in this block

Reply:
No, as this are different `name`'s and putting it in a closure breaks the lifetime extension (could replace with `Cow` as in other places in this PR but seems like something for a different PR).